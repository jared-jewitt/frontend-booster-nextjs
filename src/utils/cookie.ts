interface Options {
  path?: string;
  domain?: string;
  secure?: boolean;
  days?: number;
}

export const cookie = {
  get(name: string): string {
    const cookie = document.cookie.match(`(?:(?:^|.*; *)${name} *= *([^;]*).*$)|^.*$`)[1];
    if (cookie) return decodeURIComponent(cookie);
  },
  set(name: string, value: string | number | boolean, options: Options = {}): void {
    if (options.days) {
      options["max-age"] = options.days * 60 * 60 * 24;
      delete options.days;
    }

    document.cookie =
      name + "=" + encodeURIComponent(value) + Object.entries(options).reduce((acc, [k, v]) => `${acc}; ${k}=${v}`, "");
  },
  delete(name: string): void {
    this.set(name, "", { days: -1 });
  },
};
