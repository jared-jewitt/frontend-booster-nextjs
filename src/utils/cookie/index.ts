interface IOptions {
  path?: string;
  domain?: string;
  secure?: boolean;
  days: number;
}

export const Cookie = {
  get(name: string): string {
    const cookie = document.cookie.match(`(?:(?:^|.*; *)${name} *= *([^;]*).*$)|^.*$`)[1];
    if (cookie) return decodeURIComponent(cookie);
  },
  set(name: string, value: string | number | boolean, options: IOptions): void {
    options["max-age"] = options.days * 60 * 60 * 24;
    delete options.days;

    document.cookie =
      name + "=" + encodeURIComponent(value) + Object.entries(options).reduce((acc, [k, v]) => `${acc}; ${k}=${v}`, "");
  },
  delete(name: string, options?: Pick<IOptions, "path" | "domain">): void {
    this.set(name, "", { ...options, days: -1 });
  },
};
