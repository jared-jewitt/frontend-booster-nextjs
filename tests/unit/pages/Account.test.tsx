import React from "react";
import Account from "@/pages/account";
import { Cookie } from "@/utils";
import { render } from "../../utils";

describe("Account page", () => {
  beforeAll(() => {
    Cookie.set("isAuthenticated", true, { path: "/", days: 7 });
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Account />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
