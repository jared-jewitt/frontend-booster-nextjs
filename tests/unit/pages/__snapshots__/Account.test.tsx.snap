// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="root undefined"
    >
      <a
        href="/login"
      >
        Login
      </a>
    </div>
    <main
      class="main root"
    >
      <p>
        My Account
      </p>
      <ul>
        <li>
          Email: foo.bar@baz.com
        </li>
        <li>
          Display name: Foo Bar Baz
        </li>
        <li>
          Date of Birth: 1990/01/01
        </li>
      </ul>
    </main>
  </div>
</DocumentFragment>
`;
